### ISSUE TITLE
Error while fetching config

### ISSUE THREAD

**Issue by tvoretsmira** [Arriven/db1000n#435]
## Expected Behavior
The config is expected to be fetched.

## Actual Behavior
2022/03/26 10:47:04.100470 config.go:58: Failed to fetch config from "https://ra
w.githubusercontent.com/db1000n-coordinators/LoadTestConfig/main/config.v0.7.jso
n": Get "https://raw.githubusercontent.com/db1000n-coordinators/LoadTestConfig/m
ain/config.v0.7.json": context deadline exceeded

## Steps to Reproduce the Problem

  1. run db1000n_0.8.8_windows_amd64\db1000n.exe
  2.
  3.

## Specifications

  - Version: 0.8.8
  - Platform: Windows 7 x64 on virtual machine
  - Subsystem:

**Comment by Arriven** [comment]
Can you check if you can access that URL in browser or with curl? Most likely this is a connection problem (could be caused by VPN, I can point you to possible workarounds in that case)

**Comment by tvoretsmira** [comment]
Upgraded to v0.8.9, issue persists. 
I cannot access URL from Google Chrome when VPN is turned on. But I can do it if VPN is off. 
I have tried another country in VPN. Google Chrome was able to load URL after looong wait. And db1000n.exe is producing new type messages:
2022/03/26 15:33:05.237031 config.go:58: Failed to fetch config from "https://ra
w.githubusercontent.com/db1000n-coordinators/LoadTestConfig/main/config.v0.7.jso
n": Get "https://raw.githubusercontent.com/db1000n-coordinators/LoadTestConfig/m
ain/config.v0.7.json": dial tcp 185.199.111.133:443: i/o timeout

2022/03/26 15:34:57.146432 config.go:58: Failed to fetch config from "https://ra
w.githubusercontent.com/db1000n-coordinators/LoadTestConfig/main/config.v0.7.jso
n": Get "https://raw.githubusercontent.com/db1000n-coordinators/LoadTestConfig/m
ain/config.v0.7.json": dial tcp: lookup raw.githubusercontent.com: no such host

**Comment by michaelKaefer** [comment]
Same here (using not Windows but Ubuntu):
```console
$ docker run --rm -it --pull always ghcr.io/arriven/db1000n
latest: Pulling from arriven/db1000n
Digest: sha256:5008f4b828091afbf4d33aafa7d7aa5d8c0caecbcee8df3088c0c5d61572a5a8
Status: Image is up to date for ghcr.io/arriven/db1000n:latest
2022/03/27 10:20:43.771119 main.go:51: DB1000n [Version: v0.8.10][PID=1]
2022/03/27 10:20:43.771318 countrychecker.go:46: Checking IP address, attempt #1
2022/03/27 10:20:46.772057 countrychecker.go:98: Can't check users country. Please manually check that VPN is enabled or that you have non Ukrainian IP address.
2022/03/27 10:20:47.772378 countrychecker.go:46: Checking IP address, attempt #2
2022/03/27 10:20:50.772751 countrychecker.go:98: Can't check users country. Please manually check that VPN is enabled or that you have non Ukrainian IP address.
2022/03/27 10:20:51.772939 countrychecker.go:46: Checking IP address, attempt #3
2022/03/27 10:20:54.775135 countrychecker.go:98: Can't check users country. Please manually check that VPN is enabled or that you have non Ukrainian IP address.
2022/03/27 10:21:14.832729 config.go:58: Failed to fetch config from "https://raw.githubusercontent.com/db1000n-coordinators/LoadTestConfig/main/config.v0.7.json": Get "https://raw.githubusercontent.com/db1000n-coordinators/LoadTestConfig/main/config.v0.7.json": context deadline exceeded
2022/03/27 10:21:15.780133 config.go:135: Decrypted config
2022/03/27 10:21:15.846643 runner.go:103: New config received, applying
2022/03/27 10:21:15.846752 runner.go:114: Config is encrypted, disabling logs
2022/03/27 10:21:15.848336 runner.go:208: 24 job instances (re)started
[Error] No traffic generated. If you see this message a lot - contact admins
2022/03/27 10:22:44.780434 config.go:58: Failed to fetch config from "https://raw.githubusercontent.com/db1000n-coordinators/LoadTestConfig/main/config.v0.7.json": Get "https://raw.githubusercontent.com/db1000n-coordinators/LoadTestConfig/main/config.v0.7.json": context deadline exceeded
2022/03/27 10:22:44.800847 runner.go:121: The config has not changed. Keep calm and carry on!
[Error] No traffic generated. If you see this message a lot - contact admins
2022/03/27 10:23:44.777684 config.go:58: Failed to fetch config from "https://raw.githubusercontent.com/db1000n-coordinators/LoadTestConfig/main/config.v0.7.json": Get "https://raw.githubusercontent.com/db1000n-coordinators/LoadTestConfig/main/config.v0.7.json": context deadline exceeded
2022/03/27 10:23:44.799319 runner.go:121: The config has not changed. Keep calm and carry on!
[Error] No traffic generated. If you see this message a lot - contact admins
```

I can access the URL with VPN enabled:
```console
$ curl -I https://raw.githubusercontent.com/db1000n-coordinators/LoadTestConfig/main/config.v0.7.json
HTTP/2 200 
...
```

**Comment by Arriven** [comment]
@michaelKaefer you can try running the container with --network host or you can try a solution with one instance of db1000n running as updater publishing local copy of the config fetching it without vpn, but that's more complex

**Comment by Arriven** [comment]
@tvoretsmira I'll try to write a quick command that would allow you to use a set of http proxies instead of relying on VPN but that might be less effective

**Comment by tvoretsmira** [comment]
A lot depends on VPN. v0.8.10 with VPN in China does not display any config fetching error.

**Comment by bitshape** [comment]
@tvoretsmira @michaelKaefer try following this example: https://github.com/Arriven/db1000n/blob/main/examples/docker/static-docker-compose.yml

This is exactly why I added updater mode to reliably bypass this issue.

**Comment by tvoretsmira** [comment]
How do I use this file?

**Comment by Arriven** [comment]
@tvoretsmira as of v0.8.12 you can use this command without VPN to use `porthole-ascend-cinnamon` proxylist instead:
`go run main.go --proxy '{{ join (split (get_url "https://raw.githubusercontent.com/porthole-ascend-cinnamon/proxy_scraper/main/proxies.txt") "\n") "," }}' --debug`

You can also replace the url with any other that would return a text file with a list of proxies but I've heard that the author of that list is happy to give it to people (see #424 )

Using VPN would be more effective as you can't run all the attacks through proxy but that's better than not being able to run any attacks at all

**Comment by Arriven** [comment]
obviously you should use `./db1000n.exe` instead of `./db1000n` on windows and can use it with docker run by just appending that `--proxy` flag to the end of the appropriate command

**Comment by tvoretsmira** [comment]
Current version of docker does not seem to support Win7 x64. I had no success installing it.
I'm still able to find VPN that causes no problems. Unfortunately, those VPN are not in Russia or Belarus. You wrote that VPN is more effective than proxy. Does it remain more effective when other countries are used?

**Comment by Arriven** [comment]
yes, in terms of effectiveness I'd put them in the following order:
1. VPN into "friendly" country
2. VPN into any country other than Ukraine
3. "Friendly" Proxies
4. Any non-Ukrainian proxies
5. No VPN or proxy at all (Ukrainian IP address)

**Comment by vadyochik** [comment]
there is also a problem with outdated config - when the first github's url has failed for some reason, the other 2 bitbucket's ones are not actual. The one doesn't exist any more (404 error) and the other contains old outdated targets:

warn    failed to fetch config  {"path": "https://raw.githubusercontent.com/xzyallzjx-231/iouzjla-612/main/20.json", "error": "Get \"https://raw.githubusercontent.com/xzyallzjx-231/iouzjla-612/main/20.js
on\": context deadline exceeded"}
warn    failed to fetch config  {"path": "https://bitbucket.org/uncork-vacant-shallow/exposable-ibuprofen/raw/main/20.json", "error": "error fetching config, code 404"}
info    loading config  {"path": "https://bitbucket.org/itarmy2022123/safjzd-dajkl-124/raw/main/20.json"}
info    the config has not changed. Keep calm and carry on!
